### PR TITLE
Feature/remove side effects

### PR DESCRIPTION
**Pull request by TroyAlford**
There are currently a number of functions in-use by this lib which alter the native prototypes, such as `Array.prototype` and `Object.prototype`, which is generally considered an anti-pattern.

This PR does the following:
 - refactors each of these functions, and each of their uses to avoid prototype side-effects
 - removes one function which appears to be entirely unused (`glaem_merge`)

Among other things, this prevents an issue which occurs if, for some reason, `glaemscribe` is loaded twice.

### CODE REVIEW

**Comment by TroyAlford** [comment]
Thanks for building/maintaining this project, btw. :) I'm currently working on a wrapper to expose the functionality as both a react component and as web-components. The web-component implementation, in particular, will make it pretty trivial to use glaemscribe on various sites. I'm working out an API like:
```
<queta-text lang="quenya-tengwar-classical" font="tengwar_guni_sindarin">
  Ai ! laurië lantar lassi súrinen ,
  yéni únótimë ve rámar aldaron !
  Yéni ve lintë yuldar avánier
  mi oromardi lissë-miruvóreva
  Andúnë pella , Vardo tellumar
  nu luini yassen tintilar i eleni
  ómaryo airetári-lírinen .

  Sí man i yulma nin enquantuva ?
</queta-text>
```

File: build/web/glaemscribe/js/charsets/eldamar.cst.js
Comment (TroyAlford) [inline build/web/glaemscribe/js/charsets/eldamar.cst.js:1]: Not sure what caused this file to be generated. It appears to be new. Maybe I ran the build incorrectly? I ran `ruby ./bin_dev/make_all.rb`

**Comment by TroyAlford** [comment]
It appears that this change doesn't actually work for some reason... I'm now getting an error of: `"Cannot understand constant HYPHEN_WORD_JOINER"` - which led me to notice that the latest NPM publish was Aug 8, 2019 - and there's a lot of change between then and now.

Is it possible the JS build has become broken based on those changes?

**Comment by BenTalagan** [comment]
Hi Troy, thanks a lot for your feedback, and for the PR! I do 100% agree with the original argument. This is bad practice especially in advanced projects that load multiple libs and this may interfere with other components. I'll merge this ASAP in the master after review. The idea of extending Array is probably coming from the fact that I had ported the library from ruby to JS having both source standing side by side, and tried to keep the JS syntax as close as possible to the ruby code, hence the `.each` (I just can't remember). But this really needs to be changed.

Also, as you've stated in one of your precedent email it's been a while since the last release. It's not that there hasn't been any work done this year on glaemscribe. We've been working hard with my linguist mate Bertrand on a decent english mode, and it's been very challenging, but we're not too far from the final milestone. Still, there's a lot of doc to be written and I was waiting for the english mode to be ready to pack a new release for glaemscribe. But it may still take a little while (probably a few weeks/months) since we're both busy atm, and things always get longer than expected.

So what I'll probably do is pack a minor intermediary release to have your changes in prod :-)

**Review by BenTalagan — APPROVED** [review]

**Comment by TroyAlford** [comment]
Much appreciated! I'd be happy to contribute more, btw, if its welcome.

**Pull request merged**
